### DISCUSSION TITLE
How to connect Prisma migrate to a VPC on CI?

### DISCUSSION THREAD

**Discussion by MorenoMdz** [prisma/prisma#22871] [Q&A]
### Question

We currently have a GH action that we manually trigger the migrations from, this way we do not have to share the direct URL to our production environments. But now we locked our DB to run on a VPC, this caused the GH action to not be able to access the direct URL anymore.

We cannot whitelist the GH actions IP ranges as Gh has too many ranges, also AFAIK we cannot have a GH action running on a VPC, unless I self host the GH action server myself which I rather not do if possible.

Any other suggestions?

**Comment by ludralph** [comment]
Hi @MorenoMdz 👋 

I have shared your question internally with the engineering team. We'll get back to you soon.

**Reply by ludralph** [reply]
You would need to find a way to open up the VPC to the outside. You may need to perform these steps:

1. Create a bastion server within your VPC.
2. Use SSH to tunnel into the bastion server with port mapping.
3. Run Prisma migrations through the tunnel.

After setting up the tunnel, you would configure your .env file for Prisma to use the local port that is mapped to the remote database. Another way would be to have a dedicated lambda function for this inside the VPC.

**Reply by MorenoMdz** [reply]
When you say running it from a serverless function, is it possible to run the prisma CLI from that?

**Reply by ludralph** [reply]
Yes, it's possible. You can take a look at this [comment](https://github.com/prisma/prisma/issues/2980#issuecomment-659616226).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
